src/data_hub/api.py

```diff
@@ -1089,17 +1122,24 @@ def _row_to_dict(row, *, file_ids: list):
             continue
         if field.value is None:
             value = None
-        elif field.type in ["float", "int", "boolean"]:
+        elif field.type in ["float", "integer", "boolean"]:
             value = field.value
         elif field.type == "select":
             value = field.value.selected_options
 
         elif field.type == "reference":
             value = field.value.row_ids
-
+            reference_ids.extend(value)
         elif field.type == "file":
             value = field.value.file_ids
             file_ids.extend(value)
+        elif field.type == "expression":
+            value = field.value.result
```

Comment:
Technically the `type` of an expression column is the return type specified in the column definition.

Reply:
we'll have to dig deeper into this. expressions not well supported in this PR because there are issues with them in the kitchen sink DB